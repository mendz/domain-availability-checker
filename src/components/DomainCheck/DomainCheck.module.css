.DomainCheck {
   width: 100%;
   display: flex;
   flex-flow: column;
   /* box-shadow: 0 0 3px 1px rgba(0, 0, 0, 0.5); */
   margin-top: 2rem;
   padding: 1rem;
}