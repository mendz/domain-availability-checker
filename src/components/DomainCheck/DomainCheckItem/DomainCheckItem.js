import React from 'react';
import PropTypes from 'prop-types';
import Spinner from '../../UI/Spinner/Spinner';

import classes from './DomainCheckItem.module.css';

const DomainCheckItem = ({ domainName, availability, invalid, networkError }) => {
   let isAvailable = <Spinner />;
   const availabilityClass = [classes.Availability];

   if (availability === false) {
      isAvailable = '❌';
   } else if (availability === true) {
      isAvailable = '✔';
      availabilityClass.push(classes.Green);
   }

   let name = domainName;
   let classDomainName = [classes.Name];
   if (invalid || networkError) {
      classDomainName.push(classes.Invalid);
      let errorMessage = '';
      if (invalid) {
         errorMessage = 'NOT VALID DOMAIN!';
      } else {
         errorMessage = 'NETWORK ERROR!';
      }

      name += `- ${errorMessage}`;
   }
   return (
      <div className={classes.DomainCheckItem}>
         <div className={classDomainName.join(' ')}>{name}</div>
         <div className={availabilityClass.join(' ')}>{isAvailable}</div>
      </div>
   );
};

DomainCheckItem.propTypes = {
   domainName: PropTypes.string,
   availability: PropTypes.bool,
   invalid: PropTypes.bool,
   networkError: PropTypes.bool
}

export default DomainCheckItem;