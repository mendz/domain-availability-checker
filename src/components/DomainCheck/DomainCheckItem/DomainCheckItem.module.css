.DomainCheckItem {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 5px;
  margin: 10px 0;
  box-shadow: 0px 1px 2px 1px rgba(0, 0, 0, 0.25);
  height: 64px;
  word-break: break-word;
}

@media all and (max-width: 550px) {
  .DomainCheckItem {
    flex-direction: column;
    justify-content: space-evenly;
    font-size: 0.8rem;
    flex: 1;
  }

  .Availability {
    order: 1;
  }

  .Name {
    order: 2;
  }
}

.Name {
  font-weight: bold;
  align-self: center;
  margin: 0 5px;
}

.Invalid {
  background-color: rgba(224, 148, 148, 0.315);
  color: var(--red);
}

.Availability {
  height: 22px;
  width: auto;
}

.Success {
  background-color: rgba(148, 224, 186, 0.315);
  color: var(--green);
}

/* hold the button and the item data */
.Container {
  display: flex;
  align-items: center;
}

.Container > div {
  flex: 1;
}
