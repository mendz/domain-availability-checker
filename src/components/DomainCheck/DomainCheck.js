import React from 'react';
import PropTypes from 'prop-types';
import uuidv4 from 'uuid/v4';

import DomainCheckItem from './DomainCheckItem/DomainCheckItem';

import classes from './DomainCheck.module.css';

const DomainCheck = props => {
   const list = props.listDomains
      .filter(domain => domain.name.trim() !== '')
      .map(domain => {
         let invalid = false;
         if (props.invalidDomains.find(invalidDomain => invalidDomain === domain.name)) {
            invalid = true;
         }

         return (
            <DomainCheckItem
               key={uuidv4()}
               domainName={domain.name}
               availability={domain.availability}
               invalid={invalid} />
         )
      });

   return (
      <div className={classes.DomainCheck}>
         {list}
      </div>
   )
};

DomainCheck.propType = {
   listDomains: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string,
      availability: PropTypes.bool
   })),
   invalidDomains: PropTypes.arrayOf(PropTypes.string)
}



export default DomainCheck;