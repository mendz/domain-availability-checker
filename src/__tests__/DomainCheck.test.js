import React from 'react';
import { mount } from 'enzyme';

import DomainCheck from '../components/DomainCheck/DomainCheck';
import DomainCheckItem from '../components/DomainCheck/DomainCheckItem/DomainCheckItem';

const mockDecodeDomainListArr = [
   { name: "test", availability: false, networkError: false, invalid: true },
   { name: "www.w3schools", availability: false, networkError: false, invalid: true },
   { name: "github.com", availability: false, networkError: false, invalid: false }
];

describe('<DomainCheck />', () => {
   let wrapper;

   beforeEach(() => {
      wrapper = mount(<DomainCheck listDomains={mockDecodeDomainListArr} />);
   });

   it('should have the exact number of items as state.listDomains length', () => {
      expect(wrapper.find(DomainCheckItem)).toHaveLength(wrapper.props().listDomains.length);
   });
});