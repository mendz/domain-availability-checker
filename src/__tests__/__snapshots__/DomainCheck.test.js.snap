// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DomainCheck /> matches the snapshot 1`] = `
<DomainCheck
  listDomains={
    Array [
      Object {
        "availability": false,
        "invalid": true,
        "name": "test",
        "networkError": false,
      },
      Object {
        "availability": false,
        "invalid": true,
        "name": "www.w3schools",
        "networkError": false,
      },
      Object {
        "availability": false,
        "invalid": false,
        "name": "github.com",
        "networkError": false,
      },
    ]
  }
>
  <div
    className="DomainCheck"
  >
    <DomainCheckItem
      availability={false}
      domainName="test"
      invalid={true}
      key="test-0"
      networkError={false}
    >
      <div
        className="Container"
      >
        <ButtonCopy
          copyText="test"
        >
          <Button
            className="ButtonCopy"
            clicked={[Function]}
          >
            <button
              className="Button ButtonCopy"
              onClick={[Function]}
            >
              <img
                alt="Copy"
                src="copy.svg"
              />
            </button>
          </Button>
          <Modal
            closed={[Function]}
            show={false}
          >
            <CSSTransition
              classNames={
                Object {
                  "enterActive": "ModalOpen",
                  "exitActive": "ModalClose",
                }
              }
              in={false}
              mountOnEnter={true}
              timeout={
                Object {
                  "enter": 400,
                  "exit": 1000,
                }
              }
              unmountOnExit={true}
            >
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={true}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={
                  Object {
                    "enter": 400,
                    "exit": 1000,
                  }
                }
                unmountOnExit={true}
              />
            </CSSTransition>
          </Modal>
        </ButtonCopy>
        <div
          className="DomainCheckItem Invalid"
        >
          <span
            className="Name"
          >
            <a
              href="http://test"
              rel="noopener noreferrer"
              target="_blank"
            >
              test - NOT VALID DOMAIN!
            </a>
          </span>
          <span
            className="Availability"
          >
            <SymbolsCheck
              type="error"
            >
              <span
                className="Fail"
              >
                ERROR
              </span>
            </SymbolsCheck>
          </span>
        </div>
      </div>
    </DomainCheckItem>
    <DomainCheckItem
      availability={false}
      domainName="www.w3schools"
      invalid={true}
      key="www.w3schools-1"
      networkError={false}
    >
      <div
        className="Container"
      >
        <ButtonCopy
          copyText="www.w3schools"
        >
          <Button
            className="ButtonCopy"
            clicked={[Function]}
          >
            <button
              className="Button ButtonCopy"
              onClick={[Function]}
            >
              <img
                alt="Copy"
                src="copy.svg"
              />
            </button>
          </Button>
          <Modal
            closed={[Function]}
            show={false}
          >
            <CSSTransition
              classNames={
                Object {
                  "enterActive": "ModalOpen",
                  "exitActive": "ModalClose",
                }
              }
              in={false}
              mountOnEnter={true}
              timeout={
                Object {
                  "enter": 400,
                  "exit": 1000,
                }
              }
              unmountOnExit={true}
            >
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={true}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={
                  Object {
                    "enter": 400,
                    "exit": 1000,
                  }
                }
                unmountOnExit={true}
              />
            </CSSTransition>
          </Modal>
        </ButtonCopy>
        <div
          className="DomainCheckItem Invalid"
        >
          <span
            className="Name"
          >
            <a
              href="http://www.w3schools"
              rel="noopener noreferrer"
              target="_blank"
            >
              www.w3schools - NOT VALID DOMAIN!
            </a>
          </span>
          <span
            className="Availability"
          >
            <SymbolsCheck
              type="error"
            >
              <span
                className="Fail"
              >
                ERROR
              </span>
            </SymbolsCheck>
          </span>
        </div>
      </div>
    </DomainCheckItem>
    <DomainCheckItem
      availability={false}
      domainName="github.com"
      invalid={false}
      key="github.com-2"
      networkError={false}
    >
      <div
        className="Container"
      >
        <ButtonCopy
          copyText="github.com"
        >
          <Button
            className="ButtonCopy"
            clicked={[Function]}
          >
            <button
              className="Button ButtonCopy"
              onClick={[Function]}
            >
              <img
                alt="Copy"
                src="copy.svg"
              />
            </button>
          </Button>
          <Modal
            closed={[Function]}
            show={false}
          >
            <CSSTransition
              classNames={
                Object {
                  "enterActive": "ModalOpen",
                  "exitActive": "ModalClose",
                }
              }
              in={false}
              mountOnEnter={true}
              timeout={
                Object {
                  "enter": 400,
                  "exit": 1000,
                }
              }
              unmountOnExit={true}
            >
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={true}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={
                  Object {
                    "enter": 400,
                    "exit": 1000,
                  }
                }
                unmountOnExit={true}
              />
            </CSSTransition>
          </Modal>
        </ButtonCopy>
        <div
          className="DomainCheckItem Invalid"
        >
          <span
            className="Name"
          >
            <a
              href="http://github.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              github.com
            </a>
          </span>
          <span
            className="Availability"
          >
            <SymbolsCheck
              type="fail"
            />
          </span>
        </div>
      </div>
    </DomainCheckItem>
  </div>
</DomainCheck>
`;
